### PR TITLE
Fix POS item newline break

### LINKED ISSUE #2365: Multiple paragraphs in description breaks items in POS app
**Describe the bug**
When editing or adding an item to the POS app, if the description has more than 1 paragraph, it breaks the database layout. Effectively breaking the item and creating an item per paragraph with corrupted values.

**To Reproduce the bug**
Steps to reproduce the reported bug:
1. Go to "Apps"
2. Click on `Settings` of a "Point of Sale"
3. Add or edit an item
4. Add or edit the description of the item with a text that has 3 paragraphs.
5. Click `Save Settings`

**Expected behavior**
The description of edited item has 3 paragraphs

**Actual Behavior**

2 new items with broken values appear.

**screenshot**
image

**Your BTCPay Environment (please complete the following information):**
- BTCPay Server Version:    v1.0.6.2
- Deployment Method:  https://mynodebtc.com/
- Browser: Firefox

### PR DESCRIPTION
**Pull request by Kukks**
fixes #2365

### CODE REVIEW

File: BTCPayServer/Views/Apps/TemplateEditor.cshtml [hunk 1]
```diff
@@ -31,6 +31,7 @@
                     <div v-if="anyImages" class="card-img-top border-bottom" v-bind:style="getImage(item)"></div>
                     <div class="card-body">
                         <h6 class="card-title" v-html="item.title"></h6>
+                        <p v-text="item.description" style="white-space: pre;"></p>
```
Comment (dennisreimann) [inline BTCPayServer/Views/Apps/TemplateEditor.cshtml:34]: The `white-space: pre;` leads to strange side effect :)

pre

**Review by dennisreimann — CHANGES_REQUESTED** [review]

File: BTCPayServer/Views/Apps/TemplateEditor.cshtml (hunk 1 above)
Comment (Kukks) [reply BTCPayServer/Views/Apps/TemplateEditor.cshtml]: Removed it altogether, this was just scope creep :)

**Comment by dennisreimann** [comment]
I think without proper YAML parsing we are setting us up for problems with edge cases here. For instance if the string contains `"`, it'll also break – a description like this will result in an invalid template:

```
This is "THE BEST TEA EVAR" someone said.

I tend to agree. 
```

With a YAML parser we could use [multiline YAML](https://yaml-multiline.info/) and get rid of the other hacks as well. Otherwise we need to at least patch this by escaping the quotes `\"` … not optimal.

**Comment by Kukks** [comment]
> I think without proper YAML parsing we are setting us up for problems with edge cases here. For instance if the string contains `"`, it'll also break – a description like this will result in an invalid template:
> 
> ```
> This is "THE BEST TEA EVAR" someone said.
> 
> I tend to agree. 
> ```
> 
> With a YAML parser we could use [multiline YAML](https://yaml-multiline.info/) and get rid of the other hacks as well. Otherwise we need to at least patch this by escaping the quotes `\"` … not optimal.

If anything, we should get rid of the YAML parser and switch over to JSON. YAML is a superset of JSON anyway so it shouldn't be that hard. We're also already hiding the raw editor now so any benefits of yml readability has gone a long time ago.

**Comment by Kukks** [comment]
> I think without proper YAML parsing we are setting us up for problems with edge cases here. For instance if the string contains `"`, it'll also break – a description like this will result in an invalid template:
> 
> ```
> This is "THE BEST TEA EVAR" someone said.
> 
> I tend to agree. 
> ```
> 
> With a YAML parser we could use [multiline YAML](https://yaml-multiline.info/) and get rid of the other hacks as well. Otherwise we need to at least patch this by escaping the quotes `\"` … not optimal.

I had actually initially added multiline yaml support but it ended up being messy and it affected many parts.

**Comment by dennisreimann** [comment]
+ 1 for switching to JSON.

**Comment by Kukks** [comment]
> I think without proper YAML parsing we are setting us up for problems with edge cases here. For instance if the string contains `"`, it'll also break – a description like this will result in an invalid template:
> 
> ```
> This is "THE BEST TEA EVAR" someone said.
> 
> I tend to agree. 
> ```
> 
> With a YAML parser we could use [multiline YAML](https://yaml-multiline.info/) and get rid of the other hacks as well. Otherwise we need to at least patch this by escaping the quotes `\"` … not optimal.

addressed this issue

**Review by dennisreimann — APPROVED** [review]

**Comment by NicolasDorier** [comment]
do not use `@Safe.Raw` for things that are not raw html.

File: BTCPayServer/Views/AppsPublic/PointOfSale/Cart.cshtml [hunk 2]
```diff
@@ -242,18 +242,18 @@
...
-                            @:<img class="card-img-top" src="@image" alt="Card image cap" asp-append-version="true">
+                            @:<img class="card-img-top" src="@Safe.Raw(image)" alt="Card image cap" asp-append-version="true">
                         }
                         <div class="card-body p-3">
-                            <h6 class="card-title mb-0">@item.Title</h6>
+                            <h6 class="card-title mb-0">@Safe.Raw(item.Title)</h6>
                             @if (!String.IsNullOrWhiteSpace(description))
                             {
-                                <p class="card-text">@description</p>
+                                <p class="card-text">@Safe.Raw(System.Net.WebUtility.HtmlDecode(description))</p>
```
Comment (NicolasDorier) [inline BTCPayServer/Views/AppsPublic/PointOfSale/Cart.cshtml:251]: Remove HtmlDecode, if the goal is to put HTML, just `Safe.Raw`

File: BTCPayServer/Views/AppsPublic/PointOfSale/Static.cshtml [hunk 3]
```diff
@@ -21,17 +21,23 @@
...
                     {
-                        <img class="card-img-top" src="@item.Image" alt="Card image cap" asp-append-version="true">
+                        <img class="card-img-top" src="@Safe.Raw(item.Image)" alt="Card image cap" asp-append-version="true">
                     }
                     <div class="card-body pb-0">
-                        <h5 class="card-title">@item.Title</h5>
+                        <h5 class="card-title">@Safe.Raw(item.Title)</h5>
                         @if (!String.IsNullOrWhiteSpace(item.Description))
                         {
-                            <p class="card-text">@System.Net.WebUtility.HtmlDecode(item.Description)</p>
```
Comment (NicolasDorier) [inline BTCPayServer/Views/AppsPublic/PointOfSale/Static.cshtml:34]: Remove HtmlDecode, if the goal is to put HTML, just Safe.Raw

**Comment by NicolasDorier** [comment]
Do not use `Safe.Raw` for things that are not meant to be html

**Comment by NicolasDorier** [comment]
I think it breaks a test `CanCreateAppPoS `

**Comment by Kukks** [comment]
> I think it breaks a test `CanCreateAppPoS `

Fixed, it was actually an unrelated bug I uncovered...

**Pull request merged**